src/Transformer/ElementTransformer.php

```diff
@@ -29,8 +30,12 @@ public static function fromServer(?string $dataFromServer): ?array
 
         $data = json_decode($dataFromServer);
 
-        foreach ($data as &$value) {
-            ElementTransformer::stringValuesToObjects($value);
+        if ($data !== NULL) {
+            foreach ($data as &$value) {
+                ElementTransformer::stringValuesToObjects($value);
+            }
+        } else {
+            throw new NotCallableResourceException();
         }
```

Comment:
You can get rid of the `else` statement using the return early pattern (it helps us to have the code cleaner).
```suggestion
        if ($data === null) {
            throw new NotCallableResourceException();   
        }
        
        foreach ($data as &$value) {
            ElementTransformer::stringValuesToObjects($value);
        }
```